src/ui/components/SecondaryToolbox/ReactDevTools.tsx

```diff
@@ -238,6 +296,34 @@ function ReactDevtoolsPanel({
   reactInitPoint,
 }: PropsFromRedux) {
   const theme = useSelector(getTheme);
+  const [ReactDevTools, setReactDevTools] = useState<ReactDevToolsType>(null);
+
+  const initializationRef = useRef<boolean>(false);
+
+  // Lazy-load the right version of react-devtools-inline based on embedded metadata.
+  // Eventually this lazy-loading code could be done with Suspense, but for now use an effect.
+  //
+  // TODO This lint disabling shouldn't be necessary; omitting the dependencies array (in favor of a ref) is valid.
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  useEffect(() => {
+    if (initializationRef.current) {
+      // Only initialize the DevTools UI once, once conditions have been met.
```

Comment:
> We need to initialize the DevTools UI every time currentPoint changes, otherwise the React panel will not be updated when the user jumps to a different point in the timeline.

Ah, if we are re-initializing it every time the point changes then I think I have some substantial gap in my knowledge about how React DevTools works within Replay vs in a regular browser context. (That sounds really expensive. Do we have to re-initialize?)

> For performance reasons we should then also separate requesting the protocol version from the initialization of the DevTools UI so that we don't re-request the version every time the user jumps to a different point.

Yes. This makes sense, given the above.

Thanks for sharing this info. Let me take another pass at this given this new knowledge.

Reply:
Re-reading the code, and I guess now I understand why Replay was doing a "bad thing" before and initializing a component during render:
https://github.com/RecordReplay/devtools/blob/ce03f35d4f536f24a3b9cb5200de9189bf4abfa0/src/ui/components/SecondaryToolbox/ReactDevTools.tsx#L286-L307

I thought this was a _mistake_ rather than intentional.

I would love to chat some about the constraints here to see if there's a way we could avoid recreating DevTools entirely (b'c of the cost of this) but for now I'll refactor my approach here. :)

Reply:
The reason is simple: we need to feed all recorded backend messages from the beginning of the recording to the current point to the frontend. Then, when you jump to an *earlier* point in time, we'd have to *remove* some of these messages, which isn't supported by the frontend. If the frontend had a way to reset it (so that it forgets about all the messages we already fed to it) then we wouldn't need to recreate it.

Reply:
Yeah, that makes sense (at least when stepping to an earlier point).

> If the frontend had a way to reset it (so that it forgets about all the messages we already fed to it) then we wouldn't need to recreate it.

We _could_ do that 🙂 It would be a little hacky, but so is what we're already doing.